moto/s3tables/models.py

```diff
@@ -0,0 +1,485 @@
+        )
+        if new_table_bucket.arn in self.table_buckets:
+            raise ConflictException(
+                "The bucket that you tried to create already exists, and you own it."
+            )
+        self.table_buckets[new_table_bucket.arn] = new_table_bucket
+
+        return new_table_bucket
+
+    def list_table_buckets(
+        self,
+        prefix: Optional[str] = None,
+        continuation_token: Optional[str] = None,
+        max_buckets: Optional[int] = None,
+    ) -> Tuple[List[FakeTableBucket], Optional[str]]:
+        if not max_buckets:
+            max_buckets = S3TABLES_DEFAULT_MAX_BUCKETS
+
+        all_buckets = list(
+            bucket
+            for bucket in self.table_buckets.values()
+            if (prefix is None or bucket.name.startswith(prefix))
+        )
+
+        # encode bucket arn in the continuation_token together with prefix value
+        # raise invalidcontinuationtoken if the prefix changed
+        if continuation_token:
+            # expect continuation token to be b64encoded
+            token_arn, token_prefix = (
+                b64decode(continuation_token.encode()).decode("utf-8").split("|")
+            )
+            if token_prefix and token_prefix != prefix:
+                raise InvalidContinuationToken()
+            last_bucket_index = list(b.arn for b in all_buckets).index(token_arn)
+            start = last_bucket_index + 1
+        else:
+            start = 0
+
+        buckets = all_buckets[start : start + max_buckets]
+
+        next_continuation_token = None
+        if start + max_buckets < len(all_buckets):
+            next_continuation_token = b64encode(
+                f"{buckets[-1].arn}|{prefix if prefix else ''}".encode()
+            ).decode()
+
+        return buckets, next_continuation_token
+
+    def get_table_bucket(self, table_bucket_arn: str) -> FakeTableBucket:
+        bucket = self.table_buckets.get(table_bucket_arn)
+        if not bucket:
+            raise NotFoundException("The specified bucket does not exist.")
+        return bucket
+
+    def delete_table_bucket(self, table_bucket_arn: str) -> None:
+        # make sure table exists first
+        self.get_table_bucket(table_bucket_arn)
+        self.table_buckets.pop(table_bucket_arn)
+
+    def create_namespace(self, table_bucket_arn: str, namespace: str) -> Namespace:
+        bucket = self.table_buckets.get(table_bucket_arn)
+
+        if not bucket:
+            raise NotFoundException(
+                "The request was rejected because the specified resource could not be found."
+            )
+
+        if namespace in bucket.namespaces:
+            raise ConflictException(
+                "A namespace with an identical name already exists in the bucket."
+            )
+
+        ns = Namespace(
+            namespace, account_id=self.account_id, created_by=self.account_id
+        )
+        bucket.namespaces[ns.name] = ns
+        return ns
+
+    def list_namespaces(
```

Comment:
Same here - ideally we paginate using the decorator, instead of rolling our own logic